lib/hls/hls_parser.js

```diff
@@ -1810,6 +1831,59 @@ shaka.hls.HlsParser = class {
     return streamInfo;
   }
 
+  /**
+   * Parse EXT-X-I-FRAME-STREAM-INF media tag into a Stream object.
+   *
+   * @param {shaka.hls.Tag} tag
+   * @return {!shaka.hls.HlsParser.StreamInfo}
+   * @private
+   */
+  createStreamInfoFromIframeTag_(tag) {
+    goog.asserts.assert(tag.name == 'EXT-X-I-FRAME-STREAM-INF',
+        'Should only be called on iframe tags!');
+    /** @type {string} */
+    const type = shaka.util.ManifestParserUtils.ContentType.IMAGE;
+
+    const verbatimIFramePlaylistUri = tag.getRequiredAttrValue('URI');
+    const codecs = tag.getAttributeValue('CODECS') || '';
+
+    // Check if the stream has already been created as part of another Variant
+    // and return it if it has.
+    if (this.uriToStreamInfosMap_.has(verbatimIFramePlaylistUri)) {
+      return this.uriToStreamInfosMap_.get(verbatimIFramePlaylistUri);
+    }
+
+    const language = tag.getAttributeValue('LANGUAGE');
+    const name = tag.getAttributeValue('NAME');
+
+    const characteristics = tag.getAttributeValue('CHARACTERISTICS');
+
+    const streamInfo = this.createStreamInfo_(
+        verbatimIFramePlaylistUri, codecs, type, language, /* primary= */ false,
+        name, /* channelsCount= */ null, /* closedCaptions= */ null,
+        characteristics, /* forced= */ false, /* sampleRate= */ null,
+        /* spatialAudio= */ false);
+
+    // TODO: This check is necessary because of the possibility of multiple
+    // calls to createStreamInfoFromImageTag_ before either has resolved.
```

Comment:
This comment mentions the wrong function name.

Reply:
See the next comment